llvm/lib/CAS/BuiltinCAS.cpp

```diff
@@ -65,17 +29,20 @@ Expected<CASID> BuiltinCAS::parseID(StringRef Reference) {
     return createStringError(std::make_error_code(std::errc::invalid_argument),
                              "wrong size for cas-id hash '" + Reference + "'");
 
-  // FIXME: Take parsing as a hint that the ID will be loaded and do a look up
-  // of blobs and trees, rather than always allocating space for a hash.
-  return parseIDImpl(stringToHash(Reference));
+  std::string Binary;
+  if (!tryGetFromHex(Reference, Binary))
```

Comment:
How often do we parse ids?  Not sure if this is worth worrying about or not, but beware this may change performance when you switch from the current 20-byte hashes to 32-byte blake3 hashes, because it will exceed the short string optimization boundary in libc++.

Reply:
Super rare. Only command-line arguments.

Reply:
Although I think one reason I handwrote the code instead of looking for a helper was in case it was important for perf... just didn't turn out to be.

If it becomes important, we should just add a SmallVectorImpl overload, or replace it with that, since most of the existing callers would be happy/happier with it.